tests/profiling/collector/pprof_utils.py

```diff
@@ -142,8 +142,9 @@ def parse_newest_profile(filename_prefix: str) -> pprof_pb2.Profile:
     the newest profile that has given filename prefix.
     """
     files = glob.glob(filename_prefix + ".*")
-    # Sort files by creation timestamp (oldest first, newest last)
-    files.sort(key=lambda f: os.path.getctime(f))
+    # Sort files by logical timestamp (i.e. the sequence number, which is monotonically increasing);
+    # this approach is more reliable than filesystem timestamps, especially when files are created rapidly.
+    files.sort(key=lambda f: int(f.rsplit(".", 1)[-1]))
```

Comment:
Fun thought exercise: this takes O(N*lgN) time due to full sort, but we don't really need it. We're only looking for the max element, which can be found in linear time for any given set.

* Option 1: brute-force comparison of each seq_num to the running max
* Option 2: use a max-heap, which takes linear time to construct from unordered list (and take last elem)

We'd be using linear space explicitly, but the sorting already does that under the hood.